scripts/run_integration_tests.sh

```diff
@@ -0,0 +1,186 @@
+            return_value=$(serverless invoke -f $function_name --path "./input_events/$input_event_file" --java-layer-version "$tracing_layer_version")
+
+            if [ ! -f $snapshot_path ]; then
+                # If the snapshot file doesn't exist yet, we create it
+                echo "Writing return value to $snapshot_path because no snapshot exists yet"
+                echo "$return_value" >$snapshot_path
+            elif [ -n "$UPDATE_SNAPSHOTS" ]; then
+                # If $UPDATE_SNAPSHOTS is set to true, write the new logs over the current snapshot
+                echo "Overwriting return value snapshot for $snapshot_path"
+                echo "$return_value" >$snapshot_path
+            else
+                # Compare new return value to snapshot
+                diff_output=$(echo "$return_value" | diff - $snapshot_path)
+                if [ $? -eq 1 ]; then
+                    echo "Failed: Return value for $function_name does not match snapshot:"
+                    echo "$diff_output"
+                    mismatch_found=true
+                else
+                    echo "Ok: Return value for $function_name with $input_event_name event matches snapshot"
+                fi
+            fi
+        done
+
+    done
+
+done
+set -e
+
+echo "Sleeping $LOGS_WAIT_SECONDS seconds to wait for logs to appear in CloudWatch..."
+sleep $LOGS_WAIT_SECONDS
+
+echo "Fetching logs for invocations and comparing to snapshots"
+for handler_name in "${LAMBDA_HANDLERS[@]}"; do
+    for runtime in "${RUNTIMES[@]}"; do
+        function_name="${handler_name}_${runtime}"
+        function_snapshot_path="./snapshots/logs/$function_name.log"
+
+        # Fetch logs with serverless cli
+        raw_logs=$(serverless logs -f $function_name --startTime $script_start_time --java-layer-version "$tracing_layer_version")
+
+        # Replace invocation-specific data like timestamps and IDs with XXXX to normalize logs across executions
+        logs=$(
+            echo "$raw_logs" |
+                # Filter serverless cli errors
+                sed '/Serverless: Recoverable error occurred/d' |
+                # Remove blank lines
+                sed '/^$/d' |
+                # Normalize Lambda runtime report logs
+                sed -E 's/(RequestId|TraceId|SegmentId|Duration|Memory Used|"e"): [a-z0-9\.\-]+/\1: XXXX/g' |
+                # Normalize DD APM headers and AWS account ID
+                sed -E "s/(x-datadog-parent-id:|x-datadog-trace-id:|account_id:)[0-9]+/\1XXXX/g" |
+                # Normalize timestamps in datapoints POSTed to DD
+                sed -E 's/"points": \[\[[0-9\.]+,/"points": \[\[XXXX,/g' |
+                # Strip API key from logged requests
+                sed -E "s/(api_key=|'api_key': ')[a-z0-9\.\-]+/\1XXXX/g" |
+                # Normalize minor package version so that these snapshots aren't broken on version bumps
+                sed -E "s/(dd_lambda_layer:datadog-python[0-9]+_2\.)[0-9]+\.0/\1XX\.0/g" |
```

Comment:
Might want to make this Java specific, instead of 'datadog-python'
Also, are there any other Java specific logs that should be filtered out? You should be able to run the tests twice in a row without any changes.